tests/test_transpose.py

```diff
@@ -203,7 +326,11 @@ def test_transpose(transpose):
     assert np.allclose(f_tr["flags/frac_lost"][frac_freq, :], frac_lost, rtol=1e-3)
     assert np.allclose(f_tr["flags/frac_rfi"][frac_freq, :], frac_rfi, rtol=1e-3)
     assert (f_tr["flags/dataset_id"][empty_freq, :] == NULL_DSET_ID).all()
-    assert (f_tr["flags/dataset_id"][:empty_freq, :] != NULL_DSET_ID).all()
+
+    # Check dataset ID change is present
+    # expect starting ID, null ID, and at least one gain update
+    unique_dsets = np.unique(f_tr["flags/dataset_id"][:])
+    assert len(unique_dsets) >= 3
```

Comment:
Can you change this one to `== 3`? I don't see why it would ever be different from that.